[Android] crash with no logs - Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE

**Describe the Bug**
App crash (android only) when starting or reseting a playtrack, i do not have any logs affiliated to this crash and **i only found some indication with flipper debugger**

Before i update my sdk version from 30 to 31 and i was on lib version 2.1.2 i didn't get any crash

error given : 
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.

FATAL EXCEPTION: main
Process: com.appname, PID: 15830
java.lang.RuntimeException: Unable to start service com.guichaguri.trackplayer.service.MusicService@8f5629a with Intent { cmp=com.appname/com.guichaguri.trackplayer.service.MusicService }: java.lang.IllegalArgumentException: com.appname: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5110)
	at android.app.ActivityThread.access$2200(ActivityThread.java:310)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2319)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.app.ActivityThread.main(ActivityThread.java:8663)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:567)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1135)
Caused by: java.lang.IllegalArgumentException: com.appname: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent.
Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.
	at android.app.PendingIntent.checkFlags(PendingIntent.java:382)
	at android.app.PendingIntent.getBroadcastAsUser(PendingIntent.java:673)
	at android.app.PendingIntent.getBroadcast(PendingIntent.java:660)
	at android.support.v4.media.session.MediaSessionCompat.<init>(MediaSessionCompat.java:559)
	at android.support.v4.media.session.MediaSessionCompat.<init>(MediaSessionCompat.java:530)
	at android.support.v4.media.session.MediaSessionCompat.<init>(MediaSessionCompat.java:494)
	at com.guichaguri.trackplayer.service.metadata.MetadataManager.<init>(MetadataManager.java:62)
	at com.guichaguri.trackplayer.service.MusicManager.<init>(MusicManager.java:68)
	at com.guichaguri.trackplayer.service.MusicService.onStartCommand(MusicService.java:108)
	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:5092)
	... 9 more

**Steps To Reproduce**
Use target and compileSdkVersion = 31 and set lib version to 2.1.3
after this call any function from TrackPlayerInstance


**Code To Reproduce**
Due to some complexe integration in my application (a mix between redux, sagas) i m not able to provide sample of code.

But this occur at any function call of "TrackPlayerInstance"

**Environment Info:**
System:
    OS: macOS 12.3.1
    CPU: (8) x64 Apple M1
    Memory: 24.67 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.19.1 - ~/.nvm/versions/node/v14.19.1/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v14.19.1/bin/yarn
    npm: 6.14.16 - ~/.nvm/versions/node/v14.19.1/bin/npm
    Watchman: 2022.03.21.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8309675
    Xcode: 13.3.1/13E500a - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.11 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.67.4 => 0.67.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

"react-native-track-player": "2.1.3",
Real device - galaxy s10 + - android version 12

**How I can Help**
Actually i am looking for a solution and it seems to be a recognized error on some others libs with ancient / outdated libraries but i am actually not able to provide a patch

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -88,7 +88,7 @@ dependencies {
 
     // Make sure we're using androidx
     implementation "androidx.core:core:1.3.0-rc01"
-    implementation "androidx.media:media:1.1.0"
+    implementation "androidx.media:media:1.4.1"
     implementation "androidx.localbroadcastmanager:localbroadcastmanager:1.1.0-alpha01"
     implementation "com.github.bumptech.glide:glide:4.7.1"
 }

--- a/android/src/main/java/com/guichaguri/trackplayer/service/metadata/MetadataManager.java
+++ b/android/src/main/java/com/guichaguri/trackplayer/service/metadata/MetadataManager.java
@@ -81,7 +81,12 @@ public class MetadataManager {
         openApp.setAction(Intent.ACTION_VIEW);
         openApp.setData(Uri.parse("trackplayer://notification.click"));
 
-        builder.setContentIntent(PendingIntent.getActivity(context, 0, openApp, PendingIntent.FLAG_CANCEL_CURRENT));
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.S) {
+            builder.setContentIntent(PendingIntent.getActivity(context, 0, openApp, PendingIntent.FLAG_IMMUTABLE));
+        }
+        else {
+            builder.setContentIntent(PendingIntent.getActivity(context, 0, openApp, PendingIntent.FLAG_CANCEL_CURRENT));
+        }
 
         builder.setSmallIcon(R.drawable.play);
         builder.setCategory(NotificationCompat.CATEGORY_TRANSPORT);

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "react-native-track-player",
-  "version": "2.1.3",
+  "version": "2.1.4",
   "description": "A fully fledged audio module created for music apps",
   "main": "lib/index.js",
   "types": "lib/index.d.ts",

